src/layout/layout.js

```diff
@@ -284,6 +284,11 @@
       this.element_.parentElement.removeChild(this.element_);
       container.appendChild(this.element_);
 
+      var focusedElement = this.element_.querySelector('[autofocus]');
```

Comment:
I think this is more an issue with Firefox than with mdl. 
Basically the focused element is "lost" during the layout upgrade (specifically, after the execution of lines 284/285).

A better solution might be to find the `:focus`ed (rather than the autofocused) element before line 284 and focus it after the line 285. What do you think?